# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationUtils.h
# Commit: 3daeacd08336
# Full Hash: 3daeacd0833682a474d9bd1adc8817474a7602be
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 6: Move NotificationObserver functions to NotificationUtils r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227691
# ==============================================================================

diff -r a497ddb030f6 -r 3daeacd08336 dom/notification/NotificationUtils.h
--- a/dom/notification/NotificationUtils.h	Thu Nov 07 17:32:05 2024 +0000
+++ b/dom/notification/NotificationUtils.h	Thu Nov 07 17:32:05 2024 +0000
@@ -84,6 +84,13 @@
 void UnregisterNotification(nsIPrincipal* aPrincipal, const nsString& aId,
                             const nsString& aAlertName, CloseMode aCloseMode);
 
+nsresult RemovePermission(nsIPrincipal* aPrincipal);
+nsresult OpenSettings(nsIPrincipal* aPrincipal);
+
+enum class NotificationStatusChange { Shown, Closed };
+nsresult AdjustPushQuota(nsIPrincipal* aPrincipal,
+                         NotificationStatusChange aChange);
+
 }  // namespace mozilla::dom::notification
 
 #endif  // DOM_NOTIFICATION_NOTIFICATIONUTILS_H_